code/modules/mining/ore_detector.dm

```diff
@@ -0,0 +1,115 @@
+/obj/item/ore_detector
+	name = "ore detector"
+	desc = "A device capable of locating and displaying ores to the average untrained hole explorer."
+	icon = 'icons/obj/contained_items/tools/ore_scanner.dmi'
+	icon_state = "ore_scanner"
+	item_state = "ore_scanner"
+	w_class = ITEMSIZE_SMALL
+	slot_flags = SLOT_BELT
+	force = 1
+	var/active = FALSE
+	var/mob/our_user
```

Comment:
Use a weakref. You also don't clean this up in Destroy.

Reply:
did so, it's cleaned up in deactivate which is called in destroy